samples/go/csv/csv-import/importer.go

```diff
@@ -146,15 +151,19 @@ func main() {
 		headers = strings.Split(*header, string(comma))
 	}
 
-	ds, err := spec.GetDataset(flag.Arg(dataSetArgN))
-	d.CheckError(err)
-	defer ds.Database().Close()
-
 	kinds := []types.NomsKind{}
 	if *columnTypes != "" {
 		kinds = csv.StringsToKinds(strings.Split(*columnTypes, ","))
+	} else if *detectColumnTypes {
+		kinds = csv.GetSchema(cr, headers)
```

Comment:
I think it would be slightly simpler if you just moved this code up to line 117. I think it would make it more clear that this detectColumnTypes arg causes something fundamentally different to happen and it would keep this latter code a little simpler. That's just my opinion though, so only make the change if you agree.